src/iuse_actor.cpp

```diff
@@ -3104,3 +3105,41 @@ long place_trap_actor::use( player * const p, item * const it, bool, const tripo
     }
     return 1;
 }
+
+void emit_actor::load( JsonObject &obj )
+{
+    assign( obj, "emits", emits );
+    assign( obj, "scale_qty", scale_qty );
+}
+
+long emit_actor::use( player*, item *it, bool, const tripoint &pos ) const
+{
+    const float scaling = scale_qty ? it->charges : 1;
+    for( const auto &e : emits ) {
+        g->m.emit_field( pos, e, scaling );
+    }
+
+    return 1;
+}
+
+iuse_actor *emit_actor::clone() const
+{
+    return new emit_actor( *this );
+}
+
+void emit_actor::finalize( const itype_id &my_item_type )
+{
+    /*
+    // @todo This must be called after all finalization
```

Comment:
`emit` structure has a non-standard verification. Instead of separating finalization and consistency check, it does finalization in consistency check. This means that `emit` structures can't be treated as valid at any point before the game starts.